File: backend/src/main/java/com/backend/global/security/JwtErrorResponseWriter.java
/*
package com.backend.global.security;

import com.backend.global.response.ErrorCode;
import com.fasterxml.jackson.databind.ObjectMapper;
import jakarta.servlet.http.HttpServletResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Component;

import java.io.IOException;
import java.util.Map;

@Component
@RequiredArgsConstructor
public class JwtErrorResponseWriter {

    private final ObjectMapper objectMapper;

    public void write(HttpServletResponse response, ErrorCode errorCode) throws IOException {
        response.setStatus(errorCode.getStatus().value());
        response.setContentType("application/json;charset=UTF-8");

        Map<String, Object> errorBody = Map.of(
                "code", errorCode.getCode(),
                "status", errorCode.getStatus().value(),
                "message", errorCode.getMessage()
        );

        objectMapper.writeValue(response.getWriter(), errorBody);
    }
}
*/

File: backend/src/main/java/com/backend/global/security/jwt/JwtErrorResponseWriter.kt
package com.backend.global.security.jwt

import com.backend.global.response.ErrorCode
import com.fasterxml.jackson.databind.ObjectMapper
import jakarta.servlet.http.HttpServletResponse
import org.springframework.stereotype.Component
import java.io.IOException

@Component
class JwtErrorResponseWriter(
    private val objectMapper: ObjectMapper
) {

    @Throws(IOException::class)
    fun write(response: HttpServletResponse, errorCode: ErrorCode) {
        response.status = errorCode.status.value()
        response.contentType = "application/json;charset=UTF-8"

        val errorBody = mapOf(
            "code" to errorCode.code,
            "status" to errorCode.status.value(),
            "message" to errorCode.message
        )

        objectMapper.writeValue(response.writer, errorBody)
    }

}